libraries/Zigbee/examples/Zigbee_Temp_Hum_Sensor_Sleepy/Zigbee_Temp_Hum_Sensor_Sleepy.ino

```diff
@@ -0,0 +1,125 @@
+#define TIME_TO_SLEEP  55         /* Sleep for 55s will + 5s delay for establishing connection => data reported every 1 minute */
+
+ZigbeeTempSensor zbTempSensor = ZigbeeTempSensor(TEMP_SENSOR_ENDPOINT_NUMBER);
+
+/************************ Temp sensor *****************************/
+void meausureAndSleep(){
+  // Measure temperature sensor value
+  float temperature = temperatureRead();
+
+  // Use temparture value as humidity value to demonstrate both temperature and humidity
+  float humidity = temperature;
+
+  // Update temperature and humidity values in Temperature sensor EP
+  zbTempSensor.setTemperature(temperature);
+  zbTempSensor.setHumidity(humidity);
+
+  // Report temperature and humidity values
+  zbTempSensor.reportTemperature();
+  zbTempSensor.reportHumidity();
+
+  log_d("Temperature: %.2f°C, Humidity: %.2f%", temperature, humidity);
+
+  // Put device to deep sleep
+  esp_deep_sleep_start();
+}
+
+/********************* Arduino functions **************************/
+void setup() {
+  // Init button switch
+  pinMode(BUTTON_PIN, INPUT_PULLUP);
+
+  // Configure the wake up source and set to wake up every 5 seconds
+  esp_sleep_enable_timer_wakeup(TIME_TO_SLEEP * uS_TO_S_FACTOR);
+
+  // Optional: set Zigbee device name and model
+  zbTempSensor.setManufacturerAndModel("Espressif", "SleepyZigbeeTempSensorTest");
+
+  // Set minimum and maximum temperature measurement value (10-50°C is default range for chip temperature measurement)
+  zbTempSensor.setMinMaxValue(10, 50);
+
+  // Set tolerance for temperature measurement in °C (lowest possible value is 0.01°C)
+  zbTempSensor.setTolerance(1);
+
+  // Set power source to battery and set battery percentage to measured value (now 100% for demonstration)
+  // The value can be also updated by calling zbTempSensor.setBatteryPercentage(percentage) anytime
+  zbTempSensor.setPowerSource(ZB_POWER_SOURCE_BATTERY, 100);
+
+  // Add humidity cluster to the temperature sensor device with min, max and tolerance values
+  zbTempSensor.addHumiditySensor(0,100,1);
+
+  // Add endpoint to Zigbee Core
+  Zigbee.addEndpoint(&zbTempSensor);
+
+  // Create a custom Zigbee configuration for End Device with keep alive 10s to avoid interference with reporting data
+  esp_zb_cfg_t zigbeeConfig = ZIGBEE_DEFAULT_ED_CONFIG();
+  zigbeeConfig.nwk_cfg.zed_cfg.keep_alive = 10000;
+
+  // When all EPs are registered, start Zigbee in End Device mode
+  Zigbee.begin(&zigbeeConfig, false);
+
+  // Wait for Zigbee to start
+  while(!Zigbee.isStarted()){
+    delay(100);
+  }
+
+  // Delay 5s to allow establishing connection with coordinator, needed for sleepy devices
+  delay(5000);
+}
+
+void loop() {
+  // Checking button for factory reset
+  if (digitalRead(BUTTON_PIN) == LOW) {  // Push button pressed
+    // Key debounce handling
+    delay(100);
+    int startTime = millis();
+    while (digitalRead(BUTTON_PIN) == LOW) {
+      delay(50);
+      if ((millis() - startTime) > 3000) {
```

Comment:
factoryReset time coud be a #define or const uint32_t xxx
That would make it configurable.